indexer/api/middleware/context.go

```diff
@@ -0,0 +1,52 @@
+package middleware
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/ethereum-optimism/optimism/indexer/database"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type contextKey string
+
+const (
+	loggerKey              contextKey = "logger"
+	bridgeTransfersViewKey contextKey = "bridgeTransfersView"
+)
+
+// Setters
+func setLogger(ctx context.Context, logger log.Logger) context.Context {
+	return context.WithValue(ctx, loggerKey, logger)
+}
+
+func setBridgeTransfersView(ctx context.Context, bv database.BridgeTransfersView) context.Context {
+	return context.WithValue(ctx, bridgeTransfersViewKey, bv)
+}
+
+// Getters
+func GetLogger(ctx context.Context) log.Logger {
+	logger, ok := ctx.Value(loggerKey).(log.Logger)
+	if !ok {
+		panic("Logger not found in context!")
+	}
+	return logger
+}
+
+func GetBridgeTransfersView(ctx context.Context) database.BridgeTransfersView {
+	bv, ok := ctx.Value(bridgeTransfersViewKey).(database.BridgeTransfersView)
+	if !ok {
+		panic("BridgeTransfersView not found in context!")
+	}
+	return bv
+}
+
+func ContextMiddleware(logger log.Logger, bv database.BridgeTransfersView) func(http.Handler) http.Handler {
```

Comment:
Context was cleanest way to pass down this to the handlers I could find. An alternative was to make a function that returns a handler function

Reply:
Hmm using contexts this way isn't the best go practice since it doesn't have type safety. Alternatively since handlers reside in the same package you can do something like this (Even if the the methods are in different files)

```go
type Handlers struct {
  ... shared state between handlers
}

func NewHandlers(...) {
   return Handlers{..}
}

// deposits.go
func (h Handlers) L1DepositsHandler() {
  can access shared state through `h`
}

// withdrawals.go
func (h Handlers) L2WithdrawalsHandler() {
}
```

Reply:
I made the getters and setters that get used instead of the context directly so that it is typesafe. But that felt like a hack and this feels cleaner